src/cmd/ksh93/sh/main.c

```diff
@@ -81,16 +82,6 @@ static struct stat lastmail;
 static time_t mailtime;
 static char beenhere = 0;
 
-#if _lib_sigvec
-void clearsigmask(int sig) {
-    struct sigvec vec;
-    if (sigvec(sig, NULL, &vec) >= 0 && vec.sv_mask) {
-        vec.sv_mask = 0;
```

Comment:
According to Linux man page, this specifies the signal that should be blocked in signal handler:

```
int    sv_mask;          /* Signals to be blocked in handler */
```

Reply:
Yes, but the code is rather schizophrenic about this. However, your comment caused me to realize that libast does actually depend on its `signal()` implementation. And the ksh code is both needlessly confusing given the multiple layers of redirection and, __more importantly__, in some cases is using the libast `signal()` when it should be using the ksh `sh_signal()`. As usual all the explicit casts tend to hide the problem. So I've got a new sequence of commits that address all of this.